crates/red_knot_python_semantic/src/module_resolver/path.rs

```diff
@@ -59,6 +60,27 @@ impl ModulePath {
         self.relative_path.push(component);
     }
 
+    #[must_use]
+    pub(super) fn resolve_pure_module(
+        &self,
+        resolver: &ResolverContext,
+    ) -> Option<(File, ModuleKind)> {
+        let kind = if self.relative_path.ends_with("__init__") {
+            ModuleKind::Package
+        } else {
+            ModuleKind::Module
+        };
```

Comment:
It seems to me that we can push `kind` out of this function. As far as I can tell, the kind is always `Module` for line 573 and `Package` for line 578